tsl/src/fdw/scan_exec.c

```diff
@@ -152,6 +152,26 @@ create_data_fetcher(ScanState *ss, TsFdwScanState *fsstate)
 		}
 	}
 
+	/*
+	 * Row-by-row fetcher uses COPY statement that don't work with prepared
+	 * statements. If this plan is parameterized, this means we'll have to
+	 * revert to cursor fetcher.
+	 */
+	if (num_params > 0 && fsstate->planned_fetcher_type == RowByRowFetcherType)
+	{
+		if (ts_guc_remote_data_fetcher == AutoFetcherType)
+		{
+			fsstate->planned_fetcher_type = CursorFetcherType;
+		}
+		else
+		{
+			ereport(ERROR,
+					(errmsg("cannot use row-by-row fetcher because the plan is parameterized"),
+					 errhint("use `set timescaledb.remote_data_fetcher = 'cursor'` to choose the "
+							 "fetcher type automatically")));
```

Comment:
You have two options here: either select it explicitly or let TimescaleDB select it. Also, Hints use complete sentences (https://www.postgresql.org/docs/current/error-style-guide.html).

```suggestion
					(errmsg("cannot use row-by-row fetcher because the plan is parameterized"),
					 errhint("Set \"timescaledb.remote_data_fetcher\" to \"cursor\" to explicitly set the fetcher type or use \"auto\" to select the fetcher type automatically.")));
```